src/main/java/org/apache/maven/plugins/clean/Cleaner.java

```diff
@@ -115,9 +135,102 @@ public void delete( File basedir, Selector selector, boolean followSymlinks, boo
 
         File file = followSymlinks ? basedir : basedir.getCanonicalFile();
 
+        if ( selector == null && !followSymlinks && fastDir != null )
+        {
+            // If anything wrong happens, we'll just use the usual deletion mechanism
+            if ( fastDelete( file ) )
+            {
+                return;
+            }
+        }
+
         delete( file, "", selector, followSymlinks, failOnError, retryOnError );
     }
 
+    private boolean fastDelete( File baseDir )
+    {
+        // Handle the case where we use ${maven.multiModuleProjectDirectory}/target/.clean for example
+        if ( fastDir.getAbsolutePath().startsWith( baseDir.getAbsolutePath() ) )
+        {
+            try
+            {
+                File tmpDir = createTempDir( baseDir.getParentFile(), baseDir.getName() + "." );
+                try
+                {
+                    Files.move( baseDir.toPath(), tmpDir.toPath(), StandardCopyOption.ATOMIC_MOVE );
+                    if ( session != null )
+                    {
+                        session.getRequest().getData().put( LAST_DIRECTORY_TO_DELETE, baseDir );
+                    }
+                    baseDir = tmpDir;
+                }
+                catch ( IOException e )
+                {
+                    tmpDir.delete();
+                    if ( logDebug != null )
+                    {
+                        logDebug.log( "Unable to fast delete directory: " + e );
```

Comment:
Don't do this, pass the exception directly: `"Unable to fast delete directory".  e`